packages/pieces/community/pinterest/src/lib/actions/create-pin.ts

```diff
@@ -0,0 +1,118 @@
+import { createAction, Property } from '@activepieces/pieces-framework';
+import { makeRequest } from '../common';
+import { pinterestAuth } from '../common/auth';
+import { HttpMethod } from '@activepieces/pieces-common';
+import {
+  adAccountIdDropdown,
+  boardIdDropdown,
+  boardSectionIdDropdown,
+  pinIdMultiSelectDropdown,
+} from '../common/props';
+
+export const createPin = createAction({
+  auth: pinterestAuth,
+  name: 'createPin',
+  displayName: 'Create Pin',
+  description: 'Upload an image or video to create a new Pin on a board.',
+  props: {
+    ad_account_id: adAccountIdDropdown,
+    board_id: boardIdDropdown,
+    board_section_id: boardSectionIdDropdown,
+    title: Property.ShortText({
+      displayName: 'Title',
+      required: true,
+      description: 'The title of the pin.',
+    }),
+    description: Property.LongText({
+      displayName: 'Description',
+      required: false,
+      description: 'The description of the pin.',
+    }),
+    media_url: Property.ShortText({
+      displayName: 'Image/Video URL',
+      required: true,
+      description: 'The URL of the image or video to upload.',
+    }),
+    link: Property.ShortText({
+      displayName: 'Destination Link',
+      required: false,
+      description: 'The destination link for the pin.',
+    }),
+    dominant_color: Property.ShortText({
+      displayName: 'Pin Color',
+      description: 'Dominant pin color. Hex number, e.g. "#6E7874".',
+      required: false,
+    }),
+    alt_text: Property.ShortText({
```

Comment:
Several properties have missing or unclear descriptions:
```suggestion
     alt_text: Property.ShortText({
     displayName: 'Alt Text',  // Fix capitalization
     description: 'Alternative text for accessibility and SEO (max 500 characters)',  // Add meaningful description
     required: false,
   })
```